### PR TITLE
feature/EXLM-4-header (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by adobenitesh**
**Jira ID:  (Main Story - EXLM-4)**
https://jira.corp.adobe.com/browse/EXLM-235
https://jira.corp.adobe.com/browse/EXLM-237

**Test URLs:** 

- Before: https://main--exlm--adobe-experience-league.hlx.page/fragments/en/header/topnav
- After: https://feature-EXLM-4-header--exlm--adobe-experience-league.hlx.page/fragments/en/header/topnav

### CODE REVIEW

File: blocks/header/header.js [hunk 1]
```diff
@@ -1,163 +1,159 @@
...
-
-    const classes = ['brand', 'sections', 'tools'];
-    classes.forEach((c, i) => {
-      const section = nav.children[i];
-      if (section) section.classList.add(`nav-${c}`);
+  const response = await fetch(`${CONFIG.topNavPath}`);
+
+  if (response.ok) {
+    const topNavContent = (await response.text()).trim();
+    block.innerHTML = `<div class="exl-topnav-wrapper"><nav class="exl-topnav" aria-label="Main">${topNavContent}</nav></div>`;
```
Comment (ahmed-musallam) [inline blocks/header/header.js:45]: create these with `document.createElement`
Then assign the `nav` element the inner HTML. (so you can use the nav element below)

File: blocks/header/header.js [hunk 2]
```diff
@@ -1,163 +1,159 @@
...
+    const exlLogo = wrapper.querySelector(
+      '.exl-topnav-wrapper .exl-topnav > div:nth-child(1)',
+    );
+    exlLogo.className = 'exl-brand-container';
+
+    // Remove extra Div blocks from logo block
+    removeExtraDivs(exlLogo, 'h2');
+
+    // Assign class identifiers to parent div blocks
+    const selectors = [
```
Comment (ahmed-musallam) [inline blocks/header/header.js:57]: Can you explain the need for this? seems extremely verbose.
I'd suggest getting the `nav` element, then getting it's children, then doing whatever work you need on the children, like assigning them classes: `exl-navAction` and `exl-navItem large-menu` and so on

File: blocks/header/header.js (hunk 1 above)
Comment (ahmed-musallam) [reply blocks/header/header.js:45]: this entire HTML comes from the converter, as is, why not assign classes there?

**Review by ahmed-musallam — CHANGES_REQUESTED** [review]

File: blocks/header/header.scss [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+              transition: transform 0.1s ease;
+              top: 50%;
+              width: 6px;
+            }
+          }
+        }
+      }
+    }
+
+    &-navAction {
```
Comment (ahmed-musallam) [inline blocks/header/header.scss:104]: can me make this `nav-action`

File: blocks/header/header.scss [hunk 4]
```diff
@@ -0,0 +1,158 @@
+.header {
+  * {
+    box-sizing: border-box;
+    margin: 0;
+    padding: 0;
+  }
+
+  .exl {
+    &-topnav {
```
Comment (ahmed-musallam) [inline blocks/header/header.scss:9]: Can we avoid this nesting partial class names, this is hard to follow.

make it `exl-topnav` and do the same for other classes

**Review by ahmed-musallam — CHANGES_REQUESTED** [review]

File: blocks/header/header.js (hunk 1 above)
Comment (adobenitesh) [reply blocks/header/header.js]: Done

File: blocks/header/header.scss (hunk 3 above)
Comment (adobenitesh) [reply blocks/header/header.scss]: Done

File: blocks/header/header.scss (hunk 4 above)
Comment (adobenitesh) [reply blocks/header/header.scss]: Done

File: blocks/header/header.js (hunk 2 above)
Comment (adobenitesh) [reply blocks/header/header.js:66]: This is because only specific list of div blocks need to be given '.exl-nav-item' name hence grouping into an array and assigning selector identifier only to specific nav elements.

Comment (adobenitesh) [reply blocks/header/header.js:66]: This is because only specific list of div blocks need to be given '.exl-nav-item' name hence grouping into an array and assigning selector identifier only to specific nav elements.

File: blocks/header/header.js [hunk 5]
```diff
@@ -1,163 +1,224 @@
...
+const CONFIG = {
+  icon: {
+    aLogo:
+      '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 133.46 118.11" alt="Adobe, Inc."><defs><style>.cls-1{fill:#fa0f00;}</style></defs><polygon class="cls-1" points="84.13 0 133.46 0 133.46 118.11 84.13 0"></polygon><polygon class="cls-1" points="49.37 0 0 0 0 118.11 49.37 0"></polygon><polygon class="cls-1" points="66.75 43.53 98.18 118.11 77.58 118.11 68.18 94.36 45.18 94.36 66.75 43.53"></polygon></svg>',
+    language:
+      '<img src="https://experienceleague.adobe.com/assets/img/globegrid.svg" alt="Select Your Language" />',
+  },
+  isDesktop: window.matchMedia('(min-width: 900px)'),
+  basePath:
+    'https://51837-exlmconverter-dev.adobeioruntime.net/api/v1/web/main/convert/fragments/en/header',
```
Comment (ahmed-musallam) [inline blocks/header/header.js:11]: You cannot use the converter path here, you have to get the HTML content via `/fragments/en/header.plain.html`

Converter path can change, and we should not be calling the converter directly anyway, all HTML requests should go through franklin

**Review by ahmed-musallam — CHANGES_REQUESTED** [review]

**Comment by ahmed-musallam** [comment]
@niteshadobe please take a look at the review items, we cannot call the converter directly

**Comment by adobenitesh** [comment]
Closing this PR as another PR for similar task already merged.
https://github.com/adobe-experience-league/exlm/pull/55
